packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -271,22 +297,48 @@ class AndroidStudio implements Comparable<AndroidStudio> {
       });
     }
 
-    // Read all $HOME/.AndroidStudio*/system/.home files. There may be several
-    // pointing to the same installation, so we grab only the latest one.
-    if (globals.fsUtils.homeDirPath != null &&
-        globals.fs.directory(globals.fsUtils.homeDirPath).existsSync()) {
-      final Directory homeDir = globals.fs.directory(globals.fsUtils.homeDirPath);
-      for (final Directory entity in homeDir.listSync(followLinks: false).whereType<Directory>()) {
-        if (!entity.basename.startsWith('.AndroidStudio')) {
-          continue;
-        }
+    // Read all $HOME/.AndroidStudio*/system/.home
```

Comment:
Because you reformatted this significantly, it is difficult to tell what is existing logic and what is new. For future PRs, you should try to keep your diff as small as possible

Reply:
The base directory location, it name and the location of the `.home` text file has changed. I had to adapt the code to work on both cases without duplication.